From cec4da23284432ae4de1bb7c9e58ec1e1be800d4 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Fri, 4 Jul 2014 10:11:28 +0800
Subject: [PATCH 4438/5965] thermal: fix compiled error when
 CONFIG_AMLOGIC_THERMAL isn't selected

In header file, the fucntion need to be defined <static>.

Otherwise, compiled error will occur as following:

	multiple definition of `thermal_firmware_init'
	multiple definition of `get_cpu_temp'
---
 include/linux/thermal.h | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)
 mode change 100644 => 100755 include/linux/thermal.h

diff --git a/include/linux/thermal.h b/include/linux/thermal.h
old mode 100644
new mode 100755
index ccc75f6605d2..78c3023fd7e8
--- a/include/linux/thermal.h
+++ b/include/linux/thermal.h
@@ -261,15 +261,16 @@ static inline int thermal_generate_netlink_event(struct thermal_zone_device *tz,
 
 #ifdef CONFIG_AMLOGIC_THERMAL
 #include <mach/thermal.h>
-#else 
-int thermal_firmware_init()
+#else
+static int thermal_firmware_init(void)
 {
 	return -1;
 }
-int get_cpu_temp(void)
+
+static int get_cpu_temp(void)
 {
 	return -1;
 }
+#endif /* CONFIG_AMLOGIC_THERMAL */
 
-#endif
 #endif /* __THERMAL_H__ */
-- 
2.19.0

